Test/VsVimTest/ReferenceTest.cs

```diff
@@ -141,6 +141,11 @@ internal static ReferenceData GetReferenceData(AssemblyName assemblyName)
                 switch (name)
                 {
                     case "System":
+                    case "System.Collections.Immutable":
+                    case "Microsoft.CodeAnalysis":
+                    case "Microsoft.CodeAnalysis.CSharp":
+                    case "Microsoft.CodeAnalysis.CSharp.Scripting":
+                    case "Microsoft.CodeAnalysis.Scripting":

```

Comment:
These are more like `ReferenceKind.Editor` than `ReferenceKind.Framework`. The `Framework` notation means it's a part of the .NET Framework and hence the version doesn't really matter. The .NET Framework takes care of all the compat concerns here as we're careful to target the correct minimum target framework in the project files. The C# scripting assemblies though have different versions on different versions of Visual Studio and we need to be careful to reference the right ones.

One big difference though is we can't make the major version of the assembly to a VS version like we can for editor assemblies. I think we need to create a new `ReferenceKind` value here and lets call it `ComponentVersioned`. The mapping from version number to VS version though is likely going to be pretty specific. It can't be generalized. The best way can do is map major versions of Microsoft.CodeAnalysis (2 -> 2017, 3 -> 2019). For System.Collections.Immutable we'll likely just need to just map the specific versions.

Reply:
This is a weird area. If you get stuck on this let me know. It may be faster for me to just do the change locally and push an update to your PR.

---
In reply to: [259923068](https://github.com/jaredpar/VsVim/pull/2371#discussion_r259923068) [](ancestors = 259923068)

Reply:
I added a reference test for C# API.